search-processors/src/main/java/org/opensearch/searchpipelines/questionanswering/generative/GenerativeSearchResponse.java

```diff
@@ -57,7 +68,16 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         innerToXContent(builder, params);
         /* start of ext */ builder.startObject(EXT_SECTION_NAME);
         /*   start of our stuff */ builder.startObject(GenerativeQAProcessorConstants.RESPONSE_PROCESSOR_TYPE);
-        /*     body of our stuff    */ builder.field(GENERATIVE_QA_ANSWER_FIELD_NAME, this.answer);
+        if (answer == null) {
+            builder.field(GENERATIVE_QA_ERROR_FIELD_NAME, this.errorMessage);
+        } else {
+            /*     body of our stuff    */
+            builder.field(GENERATIVE_QA_ANSWER_FIELD_NAME, this.answer);
```

Comment:
Apology if I missed it, don't we need corresponding parser?

Reply:
There is a `TODO` at the top of the file to address your question.